syntax = "proto3";

package order;

// OrderService handles order creation and retrieval
service OrderService {
  // CreateOrder creates a new order for a product
  rpc CreateOrder(CreateOrderRequest) returns (CreateOrderResponse);

  // GetOrder retrieves an order by ID
  rpc GetOrder(GetOrderRequest) returns (Order);

  // ListOrders lists recent orders
  rpc ListOrders(ListOrdersRequest) returns (ListOrdersResponse);
}

// CreateOrderRequest is the request to create a new order
message CreateOrderRequest {
  int32 product_id = 1;
  int32 quantity = 2;
}

// CreateOrderResponse is the response after creating an order
message CreateOrderResponse {
  int32 order_id = 1;
  string status = 2;
}

// GetOrderRequest is the request to get an order by ID
message GetOrderRequest {
  int32 order_id = 1;
}

// Order represents a complete order record
message Order {
  int32 id = 1;
  int32 product_id = 2;
  int32 quantity = 3;
  string status = 4;
  string created_at = 5;
}

// ListOrdersRequest is the request to list recent orders
message ListOrdersRequest {
  int32 limit = 1;
}

// ListOrdersResponse contains a list of orders
message ListOrdersResponse {
  repeated Order orders = 1;
}
